### PR TITLE
RSDK-5658 expand support for URDF files

### PR DESCRIPTION
**Pull request by raybjork**
This PR adds functionality to our support of URDF files, allowing us to convert from our internal representation to the URDF version

### CODE REVIEW

File: spatialmath/geometry.go [hunk 1]
```diff
@@ -66,31 +68,31 @@ type GeometryConfig struct {
 }
 
 // NewGeometryConfig creates a config for a Geometry from an offset Pose.
-func NewGeometryConfig(gc Geometry) (*GeometryConfig, error) {
+func NewGeometryConfig(g Geometry) (*GeometryConfig, error) {
```
Comment (raybjork) [inline spatialmath/geometry.go:71]: Made some changes in this file to rename variables to be more accurate as we no longer have the `gc` (GeometryCreator) type

**Review by biotinker — CHANGES_REQUESTED** [review]
IMO the URDF stuff should be in its own package somewhere, not in `spatialmath`. They don't implement any spatialmath interfaces, and introduces a specific file format to a library without file formats.

File: referenceframe/urdf.go [hunk 2]
```diff
@@ -24,57 +22,31 @@ type URDFConfig struct {
 
 // URDFLink is a struct which details the XML used in a URDF link element.
 type URDFLink struct {
-	XMLName   xml.Name `xml:"link"`
-	Name      string   `xml:"name,attr"`
-	Collision []struct {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/urdf.go:29]: We do, I just moved it to its own object.  Its easier to work with these objects when they are flatter

File: spatialmath/box.go [hunk 3]
```diff
@@ -483,3 +484,16 @@ func transformPointsToPose(facePoints []r3.Vector, pose Pose) []r3.Vector {
 	}
 	return transformedVectors
 }
+
+type urdfBox struct {
+	XMLName xml.Name `xml:"box"`
+	Size    string   `xml:"size,attr"` // "x y z" format, in meters
+}
+
+func newURDFBox(b *box) *urdfBox {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply spatialmath/box.go]: Refactored

File: utils/math.go [hunk 4]
```diff
@@ -11,6 +11,16 @@ import (
 	"gonum.org/v1/gonum/stat/distuv"
 )
 
+// MetersToMM converts a value in meters to a value in mm.
+func MetersToMM(m float64) float64 {
+	return m * 1000
+}
+
+// MMToMeters converts a value in mm to a value in meters.
+func MMToMeters(m float64) float64 {
```
Comment (biotinker) [inline utils/math.go:20]: This should probably be `mm float64`

**Review by biotinker — APPROVED** [review]
LGTM, one optional change

**Pull request merged**